src/harness/sas_testcase.py

```diff
@@ -45,3 +53,50 @@ def assertValidResponseFormatForApprovedGrant(self, grant_response):
     # operationParam should not be set if grant is approved
     self.assertFalse('operationParam' in grant_response)
     self.assertTrue(grant_response['channelType'] in ('PAL', 'GAA'))
+
+  def assertRegistered(self, registration_request,
+                       conditional_registration_data=None):
+    for device in registration_request:
+      self._sas_admin.InjectFccId({'fccId': device['fccId']})
+    if conditional_registration_data:
+      self._sas_admin.PreloadRegistrationData(conditional_registration_data)
+
+    request = {'registrationRequest': registration_request}
+    response = self._sas.Registration(request)['registrationResponse']
+
+    # Check the registration response; collect CBSD IDs
+    cbsd_ids = []
+    for resp in response:
+      self.assertEqual(resp['response']['responseCode'], 0)
+      self.assertTrue('cbsdId' in resp)
+      cbsd_ids.append(resp['cbsdId'])
+
+    # Return list of cbsd_ids
+    return cbsd_ids
+
+  def assertRegisteredAndGranted(self, registration_request, grant_request,
+                                 conditional_registration_data=None):
+    self.assertEqual(len(registration_request), len(grant_request))
+    cbsd_ids = self.assertRegistered(registration_request,
+                                     conditional_registration_data)
+    list_of_cbsd_ids_and_grant_ids = []
+
+    for cbsd_id, grant_req in zip(cbsd_ids, grant_request):
+      grant_req['cbsdId'] = cbsd_id
+
+    request = {'grantRequest': grant_request}
+    grant_response = self._sas.Grant(request)['grantResponse']
+
+    # Check the grant response; collect CBSD and Grant IDs
+    for cbsd_id, grant_resp in zip(cbsd_ids, grant_response):
```

Comment:
As assertRegisteredAndGranted is kind of a black box where it only returns the cbsd id and the grant id list. But if we want to use assertRegisteredAndGranted and assertValidResponseFormatForApprovedGrant together like we can in the case of GRA_20, it won't be possible in the current implementation as it requires grantResponse as an argument. 

We can assume that if this Helper Function will successfully register and get a valid grant, then all the required fields in the grantResponse should exist and can be validated using assertValidResponseFormatForApprovedGrant.